### ISSUE TITLE
QueryParams.toUrlParameters() should treat index as unsigned

### ISSUE THREAD

**Issue by chrismiller** [Ecwid/consul-api#50]
If a negative (signed) long is passed in for the index, the request fails with a 400 error. Consul expects an unsigned long value instead.

I've hit this problem in practice when I'm trying to make blocking ConsulClient.eventList() calls. The indexes I need to supply are generated from a UUID and hence often end up as a negative value when represented as a signed long.

QueryParams.toUrlParameters() currently contains:

```java
params.add("index=" + index);
```

I think it should actually be:

```java
params.add("index=" + Long.toUnsignedString(index));
```

That assumes Java 8. Otherwise something like this should work for the conversion:

```java
  private static String toUnsignedString(long l) {
    if (l > 0) {
      return Long.toString(l);
    } else {
      long quot = (l >>> 1) / 5;
      long rem = l - quot * 10;
      return Long.toString(quot) + rem;
    }
  }
```

**Comment by spencergibb** [comment]
See #10

**Comment by spencergibb** [comment]
You'd need to have a similar conversion on the way back in as well.

**Comment by chrismiller** [comment]
Ah thanks, not sure how I missed #10. You're right that it would theoretically need to be converted on the way back too. Given the way the event endpoint works though I don't think that's a problem in practice since the ID's nd to be generated from the event's UUID, there is no ulong ID returned from Consul directly for events.

I don't think my suggested change breaks anything that isn't already broken and it'll fix the event problem without the overhead of BigInteger. If you like, in a few days I'll have time to look at this properly and will try and send a pull request that addresses the return ulong values too.

**Comment by chrismiller** [comment]
On a related note, here's the code I wrote to convert from an event ID to wait index, without using BigInteger or any other object creation:

```java
  /**
   * Converted from https://github.com/hashicorp/consul/blob/master/api/event.go#L90-L104
   * This is a hack. It simulates the index generation to convert an event ID into a WaitIndex.
   */
  private static long toIndex(String eventId) {
    long lower = 0, upper = 0;
    for (int i = 0; i < 18; i++) {
      if (i != 8 && i != 13) {
        lower = lower * 16 + Character.digit(eventId.charAt(i), 16);
      }
    }
    for (int i = 19; i < 36; i++) {
      if (i != 23) {
        upper = upper * 16 + Character.digit(eventId.charAt(i), 16);
      }
    }
    return lower ^ upper;
  }
```
 I'm wondering if it's worth adding this conversion to consul-api somewhere so users can just call Event.getWaitIndex() rather than have everyone keep reinventing the wheel? For comparison, note that Spring's EventService does the conversion using BigInteger here:

https://github.com/spring-cloud/spring-cloud-consul/blob/master/spring-cloud-consul-bus/src/main/java/org/springframework/cloud/consul/bus/EventService.java#L80-L97

**Comment by vgv** [comment]
Fixed in #51
